love/src/components/FlightTracker/FlightTracker.jsx

```diff
@@ -0,0 +1,252 @@
+    const c = 2 * Math.atan2(Math.sqrt(a), Math.sqrt(1 - a));
+
+    const d = earthRadius * c; // in km
+    return d;
+  };
+
+  componentDidUpdate = (prevProps) => {
+    if(!isEqual(prevProps.aircrafts, this.props.aircrafts)) {
+      const aircraftInRadius = this.props.aircrafts.filter((aircraft) => aircraft.id !== undefined &&
+        aircraft.id !== null && aircraft.id !== '' ).length;
+      this.setState({
+        aircraftInRadius: aircraftInRadius,
+        lastUpdate: Date.now(),
+      });
+    }
+  };
+
+  componentDidMount = () => {
+    const aircraftInRadius = this.props.aircrafts.filter((aircraft) => aircraft.id !== undefined &&
+      aircraft.id !== null && aircraft.id !== '' ).length;
+    this.setState({
+      aircraftInRadius: aircraftInRadius,
+      lastUpdate: Date.now(),
+    });
+  };
+
+  /**
+   * Function to set zoom state by the button Zoom In
+   */
+  zoomIn = () => {
+    const { zoom } = this.state;
+    if (zoom === '200') this.setState({ zoom: '160' });
+    else if (zoom === '160') this.setState({ zoom: '100' });
+  };
+
+  /**
+   * Function to set zoom state by the button Zoom Out
+   */
+  zoomOut = () => {
+    const { zoom } = this.state;
+    if (zoom === '100') this.setState({ zoom: '160' });
+    else if (zoom === '160') this.setState({ zoom: '200' });
+  };
+
+  /**
+   * Function to determine the status of the aircraft, from the distance to the radar
+   */
+  distanceStatus = (distance) => {
+    if (distance < this.RADIO2) return 'alert';
+    if (distance < this.RADIO1) return 'warning';
+    return 'running';
+  }
+
+  render() {
+    const tableData = [];
+    const headers= [
+      {
+        field: 'id',
+        title: 'AirCraft ID',
+        type: 'string',
+        render: (value) => {
+          return value ?? '-';
+        },
+      },
+      {
+        field: 'distance',
+        title: 'Distance To Center',
+        type: 'array',
+        className: styles.statusColumn,
+        render: (value) => {
+          if (value) return (
+            <StatusText small status={this.distanceStatus(value)}>
+              {value.toString() + ' km'}
+            </StatusText>
+          );
+          else return '-';
+        },
+      },
+      {
+        field: 'latitude',
+        title: 'Latitude',
+        render: (value) => {
+          if (value) return Math.round(value * 100) / 100;
+          else return '-';
+        },
+      },
+      {
+        field: 'longitude',
+        title: 'Longitude',
+        render: (value) => {
+          if (value) return Math.round(value * 1000) / 1000;
+          else return '-';
+        },
+      },
+      {
+        field: 'speed',
+        title: 'Ground Speed',
+        type: 'string',
+        render: (value) => {
+          if (value) return value + ' mph';
+          else return '-';
+        },
+      },
+    ];
+
+    const { aircrafts, status } = this.props;
+    aircrafts.forEach(element => {
+      tableData.push(element);
+    });
```

Comment:
As we are not mutating the `aircrafts` param, resulting on having `tableData` just a duplicated of the other param, I would recommend to use directly the `aircrafts` param instead of creating a duplicate.